app/models/published_vacancy.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class PublishedVacancy < Vacancy
+  # temporary solution - don't do this validation on manually published vacancies
+  validate :no_duplicate_vacancy, if: -> { external? && job_title.present? && expires_at.present? && organisation_ids.present? }
+  # this validates presence of certain fields for external vacancies
+  validates_with ExternalVacancyValidator, if: :external?
+
+  validate :enable_job_applications_cannot_be_changed_once_listed, if: -> { persisted? && listed? && enable_job_applications_changed? }
+
+  before_save :on_expired_vacancy_feedback_submitted_update_stats_updated_at, if: -> { listed_elsewhere_changed? && hired_status_changed? }
+
+  def find_conflicting_vacancy
+    find_conflict_vacancy || find_duplicate_vacancy
+  end
+
+  # soft delete so that all the stats etc are kept even after the vacancy no longer exists
+  def trash!
+    return if discarded?
+
+    supporting_documents.purge_later
+    discard!
+    remove_google_index
+  end
```

Comment:
I believe we need this `trash!` call to be defined on the 3 models.

- In the `Vacancy` model, conditionally soft/hard deleting depending on the current Vacancy `status`.
- In this `PublishedVacancy` model, as it is.
- In the `DraftVacancy` mode,l doing a hard delete.

This way, when `trash!` is called while using `Vacancy` as we do at the moment, it will soft/hard delete depending on the status. 

And when we move to not use `Vacancy` but `DraftVacancy` and `PublishedVacancy` directly, each one will behave differently upon `trash!` call.